bin/start-containter-with

```diff
@@ -2,15 +2,15 @@
 
 # docker image import https://tool.transitionmonitor.com/downloads/2diirunner_1.3.0.tar.xz 2diirunner:1.3.0
 
-analysis=$(pwd)
-data=$(dirname $(pwd))/pacta-data
-report=$(dirname $(pwd))/create_interactive_report
-stresstest=$(dirname $(pwd))/StressTestingModelDev
+analysis="$(pwd)"
+data="$(dirname $(pwd))/pacta-data"
+report="$(dirname $(pwd))/create_interactive_report"
+stresstest="$(dirname $(pwd))/StressTestingModelDev"
```

Comment:
while the quotes in `analysis="$(pwd)”` should be harmless, the more important place to put them is when you access the variable within other stuff you’re combining it with, for instance….
`data="$(dirname $(pwd))/pacta-data”`
should be
`data=$(dirname "$(pwd)")/pacta-data`

otherwise, for instance, on my machine which has a space in the working directory path, the `$(pwd)` part that gets added to the string that makes the `$data` variable only passes the path up to the first space

Reply:
I don't get why we would quote only the inner variable and not the outer one:

```bash
$(dirname "$(pwd)")
$(outer_variable "$(inner_variable)")
```

In [this example]( https://www.jimhester.com/post/2017-10-13-docker/) Jim
Hester quotes the entire expression `"/opt/$(basename $(pwd))"`

```bash
docker run -v "$(pwd)":"/opt/$(basename $(pwd))" ...
```

This is more for me to learn what's going on than a question about what to do -- which I don't mind either way.

Reply:
Oh, okay, I think I get it now. Thanks.

Reply:
it’s primarily a concern when a shell variable is expanded… that is when, if it has a space in it, often only the chunk up until the first space is expanded, so…

say you have (in pseudo-code)…
```
foo = "/path/with a/space/"
bar = "/home/$foo"

bar will return
/home/path/with

foo = "/path/with a/space"
bar = "/home/"$foo""

bar will return
/home/path/with a space/
```

Reply:
that’s actually a pretty bad example

maybe a better explanation is that in `data="$(dirname $(pwd))/pacta-data”`, `dirname $(pwd)` is a complete command of which its output is being used as a variable

so for example…
```
root@bd0b8a010eba:/bound# foo="/path with/a/space/"
root@bd0b8a010eba:/bound# echo $foo
/path with/a/space/
root@bd0b8a010eba:/bound# dirname $foo
/
with/a
root@bd0b8a010eba:/bound# dirname "$foo"
/path with/a
root@bd0b8a010eba:/bound# 
```